packaging/hip-targets-release.cmake

```diff
@@ -5,41 +5,48 @@
 # Commands may need to know the format version.
 set(CMAKE_IMPORT_FILE_VERSION 1)
 
+# Set ROCM PATH from envinorment and default to /opt/rocm if not provided.
+if( DEFINED ENV{ROCM_PATH} )
```

Comment:
An environment variable should not be used here. Users should not required to set an environment variable to use `find_package(hip)`.  The path should be computed based on the relative path of the file. Usually, its computed like this:

```cmake
# Compute the installation prefix relative to this file.
get_filename_component(_IMPORT_PREFIX "${CMAKE_CURRENT_LIST_FILE}" REALPATH)
get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
```

You can adjust the number of `get_filename_component` calls based on the relative location of the file to its installation directory.

Reply:
hipcc itself use ROCM_PATH environment variable to pick the path, hence we thought it would be OK. 
I believe ROCM_PATH  in hipcc is provided for a reason, 

I guess we have different options here.

1. If we really want to remove all references of ENV and we can do it as you mentioned above and use relative path. 

2. If ROCM_PATH in hipcc is given for a purpose I would assume same design principle here also.

3. We can give option to user to control the path by env and if anyone uses hipcc without ROCM_PATH in that case will fall back to relative path to find the ROCM path.

Reply:
@paulfreddy : bin/hipcc is generic file. It should be independent of HIP or ROCm version. So we need ROCM_PATH there. However this file is specific to a certain HIP build. However since our ROCm build system does not guarantee that HIP is relative to ROCm in the build system we cannot use Paul's solution either. So we need something similar to lines 74-79 in hipcc. So ROCM_PATH is to be derived as ENV{ROCM_PATH} if defined. Else relative to HIP_PATH as long as HIP is installed as part of ROCm. Else default to /opt/rocm.

Reply:
> However since our ROCm build system does not guarantee that HIP is relative to ROCm in the build system we cannot use Paul's solution either.

My solution does work. Just using relative paths wont work. That is why I said it needs to search for its dependencies like what hcc does does [here](https://github.com/RadeonOpenCompute/hcc/blob/clang_tot_upgrade/lib/hcc-config.cmake.in#L7). In fact, hip already does this for hcc, but doesn't do this for hsa runtime.